From: Jens Axboe <axboe@kernel.dk>
Date: Mon, 28 Aug 2017 15:06:31 -0600
Subject: [PATCH] null_blk: use available 'dev' in nullb_device_power_store()
Git-commit: b3c3051220f3e2a576ba8008c4a87b7d4c8a35e8
Patch-mainline: v4.14-rc1
References: bsc#1104967,FATE#325924

We already have this pointer, no need to use to_nullb_device()
again.

Signed-off-by: Jens Axboe <axboe@kernel.dk>
Acked-by: Hannes Reinecke <hare@suse.com>
---
 drivers/block/null_blk.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/block/null_blk.c b/drivers/block/null_blk.c
index 3b5cabe374d2..bd922868a861 100644
--- a/drivers/block/null_blk.c
+++ b/drivers/block/null_blk.c
@@ -375,7 +375,7 @@ static ssize_t nullb_device_power_store(struct config_item *item,
 
 		set_bit(NULLB_DEV_FL_CONFIGURED, &dev->flags);
 		dev->power = newp;
-	} else if (to_nullb_device(item)->power && !newp) {
+	} else if (dev->power && !newp) {
 		mutex_lock(&lock);
 		dev->power = newp;
 		null_del_dev(dev->nullb);
-- 
2.16.4

